actors/builtin/multisig/multisig_actor.go

```diff
@@ -385,23 +403,27 @@ func (a Actor) approveTransaction(rt vmr.Runtime, txnID TxnID, txn *Transaction)
 
 	// add the caller to the list of approvers
 	rt.State().Transaction(&st, func() interface{} {
+		ptx, err := adt.AsMap(adt.AsStore(rt), st.PendingTxns)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pending transactions")
 		// update approved on the transaction
 		txn.Approved = append(txn.Approved, rt.Message().Caller())
-		if err := st.putPendingTransaction(adt.AsStore(rt), txnID, txn); err != nil {
+		if err := ptx.Put(txnID, txn); err != nil {
 			rt.Abortf(exitcode.ErrIllegalState, "failed to put transaction for approval: %v", err)
 		}
+		st.PendingTxns, err = ptx.Root()
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to flush pending transactions")
 		return nil
 	})
 
 	return executeTransactionIfApproved(rt, st, txnID, txn)
 }
 
-func (a Actor) getTransaction(rt vmr.Runtime, st State, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
+func (a Actor) getTransaction(rt vmr.Runtime, st State, ptx *adt.Map, txnID TxnID, proposalHash []byte, checkHash bool) *Transaction {
 	var txn Transaction
 
 	// get transaction from the state trie
 	var err error
-	txn, err = st.getPendingTransaction(adt.AsStore(rt), txnID)
+	txn, err = getPendingTransaction(ptx, txnID, st.PendingTxns)
```

Comment:
This is very confusing. Isn't `ptx` the same map as `st.PendingTxns`? But possibly one of them is stale now? `st` should not be a parameter if we're passing the adts (and also !!! it wasn't a pointer)